### PR TITLE
enabling file permissions

### PR DESCRIPTION
**Pull request by shimonewman**
- Enabling file permissions validation via cache preview.
- Cache buster added to the cache key.
- Dropped cache usage from avater::initials

### CODE REVIEW

File: app/controllers/shared/api.php [hunk 1]
```diff
@@ -208,19 +208,52 @@
...
+
+                    if ($fileSecurity && !$valid) {
+                        $file = $dbForProject->getDocument('bucket_' . $bucket->getInternalId(), $fileId);
+                    } else {
+                        $file = Authorization::skip(fn() => $dbForProject->getDocument('bucket_' . $bucket->getInternalId(), $fileId));
+                    }
+
+                    if ($file->isEmpty()) {
+                        throw new Exception(Exception::STORAGE_FILE_NOT_FOUND);
+                    }
```
Comment (stnguyen90) [inline app/controllers/shared/api.php:250]: I tested and confirm this does work. It's just unfortunate there's a lot of duplicate code between here and the storage API.

**Review by stnguyen90 — APPROVED** [review]

**Comment by christyjacob4** [comment]
@shimonewman  we need to remove caching from the avatars API we talked about

**Comment by shimonewman** [comment]
> @shimonewman we need to remove caching from the avatars API we talked about

Done

**Review by TorstenDittmann — APPROVED** [review]

**Comment by shimonewman** [comment]
No, all good here.

[Setup] - Server database init started...
[Setup] - Creating database: appwrite...
[Setup] - Creating collection: databases...
[Setup] - Creating collection: attributes...
[Setup] - Creating collection: indexes...
[Setup] - Creating collection: projects...
[Setup] - Creating collection: platforms...
[Setup] - Creating collection: domains...
[Setup] - Creating collection: keys...
[Setup] - Creating collection: webhooks...
[Setup] - Creating collection: users...
[Setup] - Creating collection: tokens...
[Setup] - Creating collection: sessions...
[Setup] - Creating collection: teams...
[Setup] - Creating collection: memberships...
[Setup] - Creating collection: functions...
[Setup] - Creating collection: deployments...
[Setup] - Creating collection: builds...
[Setup] - Creating collection: executions...
[Setup] - Creating collection: certificates...
[Setup] - Creating collection: buckets...
[Setup] - Creating collection: stats...
[Setup] - Creating collection: realtime...
[Setup] - Creating collection: cache...
[Setup] - Creating collection: variables...
[Setup] - Creating default bucket...
[Setup] - Creating files collection for default bucket...
[Setup] - Server database init completed...
Server started successfully (max payload is 6,291,456 bytes)

**Pull request merged**